src/main/java/levy/art/json/RijkSearchFrame.java

```diff
@@ -0,0 +1,133 @@
+        topPanel.add(searchField);
+        topPanel.add(nextButton);
+        add(topPanel, BorderLayout.NORTH);
+
+        imagesPanel = new JPanel(new GridLayout(3, 4, 10, 10));
+        add(new JScrollPane(imagesPanel), BorderLayout.CENTER);
+
+        prevButton.addActionListener(e -> {
+            if (currentPage > 1) {
+                currentPage--;
+                searchArt();
+            }
+        });
+
+        nextButton.addActionListener(e -> {
+            currentPage++;
+            searchArt();
+        });
+
+        searchField.addActionListener(e -> {
+            currentPage = 1;
+            searchArt();
+        });
+
+        searchArt();
+    }
+
+    private void searchArt() {
+        if (disposable != null && !disposable.isDisposed()) {
+            disposable.dispose();
+        }
+
+        ApiKey apiKey = new ApiKey();
+        String keyString = apiKey.get();
+
+        if (searchField.getText().trim().isEmpty()) {
+
+            disposable = rijkService.getCollectionByPage(keyString, currentPage)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(SwingSchedulers.edt())
+                    .subscribe(
+                            this::handleResponse,
+                            Throwable::printStackTrace);
+        } else {
+
+            disposable = rijkService.getCollectionByQuery(keyString, searchField.getText(), currentPage)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(SwingSchedulers.edt())
+                    .subscribe(
+                            this::handleResponse,
+                            Throwable::printStackTrace);
+        }
+    }
+
+    private void handleResponse(RijksCollection rijksCollection) {
+        imagesPanel.removeAll();
+        ArtObject[] artObjects = rijksCollection.getArtObjects();
+        for (ArtObject art : artObjects) {
+            try {
+                URL url = new URL(art.getImageUrl());
+                ImageIcon originalIcon = new ImageIcon(url);
+
+                Image scaledImage = originalIcon.getImage().getScaledInstance(200, -1, Image.SCALE_SMOOTH);
+
+                JLabel label = new JLabel();
+                ImageIcon imageIcon = new ImageIcon(scaledImage);
+                label.setIcon(imageIcon);
+                label.setToolTipText(art.getTitle() + " by " + art.getArtist());
+
+                label.addMouseListener(new MouseAdapter() {
+                    @Override
+                    public void mouseClicked(MouseEvent e) {
+                        System.out.println("Yay you clicked me");
+                        new ImageFrame(art.getTitle(), art.getArtist(), art.getImageUrl()).setVisible(true);
+                    }
+                });
+
+                imagesPanel.add(label);
+            } catch (Exception ex) {
+                ex.printStackTrace();
+            }
+        }
+        imagesPanel.revalidate();
+        imagesPanel.repaint();
```

Comment:
What does this do?

Reply:
just a way to reformat after every search

Reply:
its not necessary i was just looking into how to organize the images and this was suggested